### PR TITLE
test: catch errors from propagation test

### PR DESCRIPTION
**Pull request by Rauno56**
# Description

One of the propagation tests would just time out if any of the assertions would fail.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Internal change (a change which is not visible to the package consumers)
- [ ] Documentation change or requires a documentation update

# How Has This Been Tested?

- [ ] Tested manually
- [x] Added automated tests

# Checklist:

- [x] Unit tests have been added/updated
- [ ] Documentation has been updated
- [ ] Change file has been generated (`npm run change:new`)

### CODE REVIEW

File: test/propagation.test.ts [hunk 1]
```diff
@@ -54,13 +49,11 @@ describe('propagation', () => {
       assert.strictEqual(carrier['x-b3-spanid'], spanId);
       assert.strictEqual(carrier['x-b3-sampled'], '1');
       assert.strictEqual(carrier['traceparent'], `00-${traceId}-${spanId}-01`);
-      done();
```
Comment (jtmalinowski) [inline test/propagation.test.ts:57]: if for some reason this callback is not run then the asserts will never be run and we might get false positives

File: test/propagation.test.ts [hunk 2]
```diff
@@ -102,15 +94,18 @@ describe('propagation', () => {
     const newContext = propagation.extract(context.active(), incomingCarrier);
     tracer.startSpan('request handler', {}, newContext).end();
 
-    spanProcessor.forceFlush().then(() => {
```
Comment (jtmalinowski) [inline test/propagation.test.ts:105]: let's make the whole test function an async one and flatten this callback, like so:
```js
await spanProcessor.forceFlush();

assert(exporter.getFinishedSpans().length == 1);
// ...

stopTracing();
```

**Review by jtmalinowski — CHANGES_REQUESTED** [review]
I added small suggestions to simplify things.

File: test/propagation.test.ts (hunk 1 above)
Comment (Rauno56) [reply test/propagation.test.ts:57]: By "callback" you mean `done`? If so the asserts in this test will be run regardless of us calling `done` or not. Do you have a code sample of a situation where the asserts would not be ran?

**Iff** the test function has a length of 1 (meaning we "ask" Mocha to make this an async test) `done` needs to be called though, if we don't add any arguments, Mocha assumes a synchronous test, which this is.

File: test/propagation.test.ts (hunk 2 above)
Comment (Rauno56) [reply test/propagation.test.ts:105]: I'll happily do that. Just didn't want to change more than I absolutely have to.

File: test/propagation.test.ts (hunk 1 above)
Comment (jtmalinowski) [reply test/propagation.test.ts:57]: By callback I mean the second argument to `context.with(spanContext, functionToRunInContext)`. I phrased it poorly.

**Review by jtmal-signalfx — APPROVED** [review]

**Pull request merged**